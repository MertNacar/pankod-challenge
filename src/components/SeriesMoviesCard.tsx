import React from "react";


interface IProp {
  image: string;
  alt: string;
  title: string;
}

const SeriesMoviesCard: React.FC<IProp> = (props) => {
  return (
    <div className="card" style={{ borderWidth: 0 }}>
      <img
        src={props.image}
        className="card-img-top img-responsive"
        alt={props.alt}
        style={{ maxHeight: 180, maxWidth: 150 }}
      />

      <p className="card-text">{props.title}</p>
    </div>
  );
};

export default SeriesMoviesCard;
