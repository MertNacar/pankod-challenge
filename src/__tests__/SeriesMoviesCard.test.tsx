import React from "react";
import { shallow } from "enzyme";
import SeriesMoviesCard from "../components/SeriesMoviesCard";
import mockData from "../../public/feed/test/mock_test.json";

describe("SeriesMoviesCard Component", () => {
  const wrapper = shallow(
    <SeriesMoviesCard
      alt={mockData[0].title}
      image={mockData[0].images["Poster Art"].url}
      title={mockData[0].title}
    />
  );

  it("is exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("return one card item filled with first item of mock data", () => {
    expect(wrapper).toMatchSnapshot("SeriesMoviesCard");
  });
});
