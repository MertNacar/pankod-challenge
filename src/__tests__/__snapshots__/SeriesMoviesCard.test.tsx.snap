// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesMoviesCard Component return one card item filled with first item of mock data: SeriesMoviesCard 1`] = `
<div
  className="card mycard-container"
>
  <img
    alt="Wolf Creek"
    className="card-img-top img-responsive"
    src="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
  />
  <div
    className="card-body mycard-body"
  >
    <p
      className="card-text mycard-image"
    >
      Wolf Creek
    </p>
  </div>
</div>
`;
